Sourcery/Parsing/FileParser.swift

```diff
@@ -573,33 +578,46 @@ extension FileParser {
         case ("="?, _?):
-             let body = wrappedBody.substring(from: wrappedBody.index(after: wrappedBody.startIndex)).trimmingCharacters(in: .whitespacesAndNewlines)
-             rawValue = parseEnumValues(body)
+            let body = wrappedBody.substring(from: wrappedBody.index(after: wrappedBody.startIndex)).trimmingCharacters(in: .whitespacesAndNewlines)
+            rawValue = parseEnumValues(body)
         case ("("?, ")"?):
-             let body = wrappedBody.substring(with: wrappedBody.index(after: wrappedBody.startIndex)..<wrappedBody.index(before: wrappedBody.endIndex)).trimmingCharacters(in: .whitespacesAndNewlines)
-             associatedValues = parseEnumAssociatedValues(body)
+            let body = wrappedBody.substring(with: wrappedBody.index(after: wrappedBody.startIndex)..<wrappedBody.index(before: wrappedBody.endIndex)).trimmingCharacters(in: .whitespacesAndNewlines)
+            associatedValues = parseEnumAssociatedValues(body)
         case (nil, nil):
             break
         default:
-             Log.warning("\(logPrefix)parseEnumCase: Unknown enum case body format \(wrappedBody)")
+            Log.warning("\(logPrefix)parseEnumCase: Unknown enum case body format \(wrappedBody)")
         }
 
+        // Parse enum case annotations
+
         let annotations: [String: NSObject]
-        if var body = extract(.keyPrefix, from: source)?.trimmingSuffix("case") {
-            // search backwards for possible enum cases separators
-            if let semicolon = body.range(of: ";", options: [.backwards])?.upperBound {
-                body = body.substring(from: semicolon)
-            } else if let comma = body.range(of: ",", options: [.backwards])?.upperBound {
-                body = body.substring(from: comma)
-            } else if let parenthesis = body.range(of: ")", options: [.backwards])?.upperBound {
-                body = body.substring(from: parenthesis)
-            } else if let `case` = body.range(of: "case", options: [.backwards])?.upperBound {
-                body = body.substring(from: `case`)
-            } else if let brace = body.range(of: "{", options: [.backwards])?.upperBound {
-                body = body.substring(from: brace)
-            }
 
-            annotations = AnnotationsParser(contents: body).all
+        var annotationRangeStart: Int64?
+        if let previousCase = `enum`.cases.last, let previousCaseKeyRange = Substring.key.range(for: previousCase.__underlyingSource) {
+            // if current enum case is not the first one in containing enum
+            // get everything after the previous case declaration
+            annotationRangeStart = previousCaseKeyRange.offset + previousCaseKeyRange.length
+        } else if let enumBody = Substring.body.range(for: `enum`.__underlyingSource) {
+            // if current enum case is the first one in containing enum
+            // get everything after enum body start
+            annotationRangeStart = enumBody.offset
```

Comment:
@krzysztofzablocki with enum cases it's a bit different because enum case can have inline annotations for associated value which can add up to the case annotations, IIRC the reason for different logic here. Also AnnotationsParser is designed to parse line by line which does not fit inline annotations. I would say that we need to revisit it at some point cause it's not good that we now have different kinds of annotations parsing logic in different places.

I do believe though this particular case can be handled with minor changes, not with reworking all the annotations parsing just for enum cases. We can just get the reference not to the previous enum case but to the previous substructure on the same level in the containing enum case, no matter if it's enum case, method, var or inner type.